src/js/akamai.js

```diff
@@ -203,16 +206,18 @@ akamai = {
    * as an object.
    */
   extractedValues(response) {
-    let pat = /^name=([^;]*); value=([^;]*).*$/;
-    return getResponseHeaderValues(response, "x-akamai-session-info")
-      .reduce(function (vars, value) {
+    let pat = /^name=(.*); value=([^\s]*)(;.*)?$/;
```

Comment:
Regex in this line is assuming that there is at most 1 ";" followed by something that needs to be ignored after the value but
- there can be 2 when variable value is extracted from qsp or from cookie. In this case variable value is not only followed by full_location_id but also by separator. In that case, the part that includes full_location_id is included in the value against which assert function is matching
- there can be 0 when value is not extracted. In this case, if variable value includes some ";", the part of this value after the last ";" matches unepectedly with the last group.

Looking closer at groups in this regex:

1. The 1st one `(.*)` is capturing but it may not need to. More importantly, it can match with any caracter including those that don't belong to a variable name. It could be more specific.
2. The 2nd one `([^\s]*)` is capturing and this is probably needed. More importantly, it excludes spaces but it shouldn't.
3. The 3rd one `(;.*)?` is capturing but it may not need to. More importantly, it could be more specific.

With this regex proposed instead `^name=([^\s]*); value=(.*)(; full_location_id=[^;]*(; separator=[^;]*)?)?$`:
- it is leaving asside capturing or not
- name and value are captured in same group 1 and 2 as before
- group 3 that captured one of full_location_id or separator before is now capturing both if present
- capture group 4 is added and it captures separator only

However, it requires the match to be ungreedy.

Attached screenshots show test results with U flag with PCRE2 for ungridiness, with sample values below:
![proposedRegex1_ungreedy_NotExtracted](https://github.com/akamai-contrib/postman-jsonnet/assets/66840378/9f3b8c53-54f7-48af-9195-db0c9cd6d708)
1. name=NOT_EXTRACTED; value= key1=value1; key2=value2
2. name=NOT_EXTRACTED; value= key3=value3

![proposedRegex1_ungreedy_ExtractedFromCookieOrQsp](https://github.com/akamai-contrib/postman-jsonnet/assets/66840378/7b69203f-06a1-449d-a78f-8712fb02ba06)
4. name=EXTRACTED_FROM_COOKIE_OR_QSP; value=key1=value1; key2=value2; full_location_id=cookieName3; separator=%3d
5. name=EXTRACTED_FROM_COOKIE_OR_QSP; value=key3=value3; full_location_id=cookieName3; separator=%3d

![proposedRegex1_ungreedy_ExtractedFromHeader](https://github.com/akamai-contrib/postman-jsonnet/assets/66840378/181b1d25-1f3e-4f59-a75a-7c576f40f6ee)
6. name=EXTRACTED_FROM_COOKIE_OR_QSP; value=key1=value1; key2=value2; full_location_id=cookieName3; separator=%3d
7. name=EXTRACTED_FROM_COOKIE_OR_QSP; value=key3=value3; full_location_id=cookieName3; separator=%3d

These sample values are meant to cover:
- variable value contains ";" or not
- variable is not extracted 
- variable is extracted from cookie (or qsp) or is extracted from header

Reply:
With a greedy match, the 1st new proposed regex capture full_location_id and separator in the group that captures value, which is not wanted.
![proposedRegex1_greedy_ExtractedFromCookieOrQsp](https://github.com/akamai-contrib/postman-jsonnet/assets/66840378/2711d2ca-c889-48a2-af0f-71a35bfa6018)

Reply:
If greediness can't be disabled, another regex similar to the one below may be used.

It it's built with an alternative where 1st option captures full_location (and eventually separator) when it's present. 2nd option should match only when full_location and separator are absent.
`^name=([^\s]*); value=(.*)(; full_location_id=.*)$|^name=([^\s]*); value=(.*)$`

A problem with this approach is that name and value are captured in group 1 and 2 when full_location_id is present
and in groups 4 and 5 otherwise. So this needs to be dealt with if captured groups are actually used.


![proposedRegex2_greedy_ExtractedFromCookieOrQsp](https://github.com/akamai-contrib/postman-jsonnet/assets/66840378/bb6efa44-cbc1-4564-952f-090c5666f707)

![proposedRegex2_greedy_ExtractedFromHeader](https://github.com/akamai-contrib/postman-jsonnet/assets/66840378/a6a05170-0ce6-45a8-9e36-a7f248b2c7c5)

![proposedRegex2_greedy_NotExtracted](https://github.com/akamai-contrib/postman-jsonnet/assets/66840378/fce84c86-5f7c-4f09-8d61-974c39cf57c4)

Reply:
In the group that captures the value of the variable, the greedy quantifier `*` can be replaced by the lazy quantifier `*?`:
`^name=([^\s]*); value=(.*?)(; full_location_id=[^;]*(; separator=[^;]*)?)?$`
![proposedRegex3_lazyQuantifier_ExtractedFromCookieOrQsp](https://github.com/user-attachments/assets/ea4451cc-b566-4d1e-b0ef-0550e50d8069)